### PR TITLE
Upgrade to Akeneo PIM 1.5

### PR DESCRIPTION
**Pull request by ctreton**
Upgrade the bundle for the 1.5 PIM

### CODE REVIEW

File: Doctrine/MongoDBODM/Filter/CompletenessFilter.php [hunk 1]
```diff
@@ -4,13 +4,13 @@
 
 use Pim\Bundle\CatalogBundle\Doctrine\MongoDBODM\ProductQueryUtility;
 use Pim\Bundle\CatalogBundle\Doctrine\MongoDBODM\Filter\CompletenessFilter as BaseCompletenessFilter;
-use Pim\Bundle\CatalogBundle\Exception\InvalidArgumentException;
+use Pim\Component\Catalog\Exception\InvalidArgumentException;
```
Comment (damien-carcel) [inline Doctrine/MongoDBODM/Filter/CompletenessFilter.php]: Move after the bundles (alphabetically ordered)

File: Processor/AttributeToFlatArrayProcessor.php [hunk 2]
```diff
@@ -17,23 +17,23 @@
...
-     * @param LocaleManager       $localeManager
+     * @param NormalizerInterface       $transNormalizer
+     * @param LocaleRepositoryInterface $localeRepository
      */
-    public function __construct(NormalizerInterface $transNormalizer, LocaleManager $localeManager)
+    public function __construct(NormalizerInterface $transNormalizer, LocaleRepositoryInterface $localeRepository)
     {
         $this->transNormalizer = $transNormalizer;
-        $this->localeManager   = $localeManager;
+        $this->localeRepository   = $localeRepository;
```
Comment (damien-carcel) [inline Processor/AttributeToFlatArrayProcessor.php]: spaces

File: DependencyInjection/PimEnhancedConnectorExtension.php [hunk 3]
```diff
@@ -30,7 +30,7 @@ public function load(array $configs, ContainerBuilder $container)
     }
 
     /**
-     * Load the mapping for product and product storage
+     * Load the mapping for product and product storage.
```
Comment (rybus) [inline DependencyInjection/PimEnhancedConnectorExtension.php]: Loads

File: Doctrine/MongoDBODM/Filter/DateTimeFilter.php [hunk 4]
```diff
@@ -3,12 +3,11 @@
...
 
 use Pim\Bundle\CatalogBundle\Doctrine\MongoDBODM\Filter\DateFilter;
-use Pim\Bundle\CatalogBundle\Exception\InvalidArgumentException;
-use Pim\Bundle\CatalogBundle\Query\Filter\Operators;
+use Pim\Component\Catalog\Exception\InvalidArgumentException;
 use Pim\Bundle\CatalogBundle\Doctrine\MongoDBODM\ProductQueryUtility;
 
 /**
- * Override of the date filter to allow the use of the time part
+ * Override of the date filter to allow the use of the time part.
```
Comment (rybus) [inline Doctrine/MongoDBODM/Filter/DateTimeFilter.php:10]: Overrides

File: Doctrine/ORM/Filter/CompletenessFilter.php [hunk 5]
```diff
@@ -4,12 +4,11 @@
...
 use Pim\Bundle\EnhancedConnectorBundle\Doctrine\ORM\Join\CompletenessJoin;
 use Pim\Bundle\CatalogBundle\Doctrine\ORM\Filter\CompletenessFilter as BaseCompletenessFilter;
-use Pim\Bundle\CatalogBundle\Exception\InvalidArgumentException;
-use Pim\Bundle\CatalogBundle\Query\Filter\FieldFilterInterface;
+use Pim\Component\Catalog\Exception\InvalidArgumentException;
 
 /**
  * Override of the completeness filter that allows to apply completeness
- * filter only on scope (channel) without locale restriction
+ * filter only on scope (channel) without locale restriction.
```
Comment (rybus) [inline Doctrine/ORM/Filter/CompletenessFilter.php:11]: Filters

**Comment by damien-carcel** [comment]
Minor comments, but GTM for me.

By the way, as you do the detach in the reader, the v1.2 could be compatible again with the PIM 1.3 and 1.4, with no need to maintain 1.1 and 1.0. To test.

File: Doctrine/ORM/Filter/CompletenessFilter.php (hunk 5 above)
Comment (fitn) [inline Doctrine/ORM/Filter/CompletenessFilter.php:11]: @rybus "... to apply completeness filter only on scope...", right?

**Comment by fitn** [comment]
As I disabled the csv family export, it can't but otherwise I agree.

**Pull request merged**